### PR TITLE
Add update fields for local and remote command

### PR DESCRIPTION
**Pull request by jkisk**
Add an optional update field to the local and remote command resources that will run on update if supplied.  If this field is omitted an update operation will re-run the create field command.

Resolves: #37

### CODE REVIEW

File: provider/pkg/provider/command.go [hunk 1]
```diff
@@ -54,6 +56,20 @@ func (c *command) RunCreate(ctx context.Context, host *provider.HostClient, urn
 	return id, err
 }
 
+//
+func (c *command) RunUpdate(ctx context.Context, host *provider.HostClient, urn resource.URN) (string, error) {
+	if c.Update != nil {
+		stdout, stderr, id, err := c.run(ctx, *c.Update, host, urn)
```
Comment (stack72) [inline provider/pkg/provider/command.go:62]: Should we dereference Update or not (we don't with Create below)

File: provider/pkg/provider/remotecommand.go [hunk 2]
```diff
@@ -124,6 +125,19 @@ func (c *remotecommand) RunDelete(ctx context.Context, host *provider.HostClient
 	return err
 }
 
+func (c *remotecommand) RunUpdate(ctx context.Context, host *provider.HostClient, urn resource.URN) (string, error) {
+	if c.Update != nil {
+		stdout, stderr, id, err := c.run(ctx, *c.Update, host, urn)
```
Comment (stack72) [inline provider/pkg/provider/remotecommand.go:130]: Same question as before

**Comment by stack72** [comment]
Would be really good to add an update test for this functionality

File: provider/pkg/provider/command.go (hunk 1 above)
Comment (jkisk) [reply provider/pkg/provider/command.go:62]: We should, because update is *string, similar to the delete field, where create is a string.

**Review by stack72 — APPROVED** [review]
Merge on a green PR run

**Comment by rbm-radius** [comment]
Yay! Thank you!

**Pull request merged**